Makefile.core.mk

```diff
@@ -63,6 +64,21 @@ mesh: vfsgen
 	go build -o $(GOPATH)/bin/mesh ./cmd/mesh.go
 	GOARCH=$(TARGET_ARCH) GOOS=$(TARGET_OS) go build -o /work/mesh ./cmd/mesh.go
 
+# NOTE: docker targets only work with local builds.
+
+controller: vfsgen
+	go build -o $(GOPATH)/bin/istio-operator ./cmd/manager
+
+docker: controller
+	mkdir -p build/out
+	cp $(GOPATH)/bin/istio-operator build/out/.
+	docker build -t $(HUB)/operator:$(TAG) -f build/Dockerfile .
+
+docker.push:
+	docker push $(HUB)/operator:$(TAG)
```

Comment:
HUB is not set in the makefile, so this may break if a user doesn't have it set? Maybe we should either set a default or fail if it is unset. We should probably not set it to `istio` to avoid accidental pushes to the official dockerhub

Reply:
It's intended to fail if HUB is unset, but it would be nice to output an error. This is also assuming there should not be a default HUB to avoid accidental pushes. @sdake is going to clean all this up to work properly in the container and address this.